src/qml/imports/Theme/QfBadge.qml

```diff
@@ -0,0 +1,28 @@
+import QtQuick
+import QtQuick.Controls
+
+Rectangle {
+  property alias badgeText: inText
+
+  anchors {
+    top: parent.top
+    right: parent.right
+    rightMargin: 2
+    topMargin: 2
+  }
+
+  width: 12
+  height: 12
+  radius: width / 2
+
+  border.width: 1.5
+  border.color: "white"
+
+  Text {
+    id: inText
+    anchors.fill: parent
+    font.pixelSize: text.length === 1 ? height / 1.6 : height / 1.8
```

Comment:
Have you tried this using a high DPI screen? the height here is not in pixel size, it's in painter unit (which will change based on screen DPI).